core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/UdtCodec.java

```
@@ -165,24 +165,45 @@ public UdtValue parse(@Nullable String value) {
     }
 
     UdtValue udt = cqlType.newValue();
+    int length = value.length();
 
     int position = ParseUtils.skipSpaces(value, 0);
-    if (value.charAt(position++) != '{') {
+    if (value.charAt(position) != '{') {
       throw new IllegalArgumentException(
           String.format(
-              "Cannot parse UDT value from \"%s\", at character %d expecting '{' but got '%c'",
+              "Cannot parse UDT value from \"%s\" at character %d: expecting '{' but got '%c'",
               value, position, value.charAt(position)));
     }
 
+    position++;
     position = ParseUtils.skipSpaces(value, position);
 
-    if (value.charAt(position) == '}') {
-      return udt;
+    if (position == length) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Cannot parse UDT value from \"%s\" at character %d: expecting CQL identifier or '}', got EOF",
+              value, position));
```

We don't have the field name yet, so I had to fail fast here.